PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/mysql/action/CreateMySQLAction.java

```diff
@@ -39,29 +36,41 @@
 import com.microsoft.intellij.AzurePlugin;
 import com.microsoft.intellij.util.AzureLoginHelper;
 import com.microsoft.tooling.msservices.components.DefaultLoader;
-import com.microsoft.tooling.msservices.helpers.Name;
-import com.microsoft.tooling.msservices.serviceexplorer.AzureIconSymbol;
+import com.microsoft.tooling.msservices.serviceexplorer.AzureActionEnum;
+import com.microsoft.tooling.msservices.serviceexplorer.Node;
 import com.microsoft.tooling.msservices.serviceexplorer.NodeActionEvent;
 import com.microsoft.tooling.msservices.serviceexplorer.NodeActionListener;
 import com.microsoft.tooling.msservices.serviceexplorer.azure.mysql.MySQLModule;
+import com.microsoft.tooling.msservices.serviceexplorer.listener.Backgroundable;
+import com.microsoft.tooling.msservices.serviceexplorer.listener.Basicable;
+import org.apache.commons.lang3.StringUtils;
 
-import static com.microsoft.intellij.ui.messages.AzureBundle.message;
+import java.util.Objects;
 
-@Name(CreateMySQLAction.ACTION_NAME)
-public class CreateMySQLAction extends NodeActionListener {
+import static com.microsoft.intellij.ui.messages.AzureBundle.message;
 
-    protected static final String ACTION_NAME = "Create";
+public class CreateMySQLAction extends NodeActionListener implements Backgroundable, Basicable {
 
     private final MySQLModule model;
+    private AzureMySQLConfig config;
 
     public CreateMySQLAction(MySQLModule model) {
         super();
         this.model = model;
     }
 
     @Override
-    public AzureIconSymbol getIconSymbol() {
-        return AzureIconSymbol.Common.CREATE;
+    public AzureActionEnum getActionEnum() {
```

Comment:
why make `AzureActionEnum`/`getActionEnum` end with `enum`?
java is a strong-type language. we know it's a enum when we call the method.

Reply:
renamed.